src/google-tag-manager.ts

```diff
@@ -83,8 +83,15 @@ export class GoogleTagManager extends BasePlugin<GoogleTagManagerConfig> {
       this.logger.warn(`'${customEvent}' is an invalid player event`);
       return;
     }
+
     this.eventManager.listen(this.player, customEvent, (event: FakeEvent) => {
-      const dataLayerVariablePayload = event.payload !== undefined ? { [`${event.type}-payload`]: event.payload } : {};
+      const eventPayload = event.payload !== undefined ? event.payload : {};
```

Comment:
you can do event.payload ? ...
unless you specifically want to test for undefined as opposed to null

Reply:
event.payload could be 0 for example, compare to undefine is safer according to yoni.